src/ssh/cockpitsshrelay.c

```diff
@@ -705,15 +684,31 @@ set_knownhosts_file (CockpitSshData *data,
         }
     }
 
-  g_debug ("%s: using known hosts file %s", data->logname, data->ssh_options->knownhosts_file);
-  if (!data->ssh_options->allow_unknown_hosts && !host_known)
+  if (!host_known && data->ssh_options->challenge_unknown_host_preconnect)
     {
-      g_message ("%s: refusing to connect to unknown host: %s:%d",
-                 data->logname, host, port);
-      problem = "unknown-host";
-      goto out;
+      authorize_knownhosts_data = challenge_for_knownhosts_data (data);
+      if (authorize_knownhosts_data)
+        {
+          if (write_tmp_knownhosts_file (data, authorize_knownhosts_data, &problem))
+            {
+              host_known = cockpit_is_host_known (tmp_knownhost_file, host, port);
+              if (host_known)
+                data->ssh_options->knownhosts_file = tmp_knownhost_file;
```

Comment:
What would cause the case where ```!host_known```? What would happen in that case? At the very least those answers should be commented here.

Reply:
This is fairly similar to the case above when we query sssd. An unknown host indeed Should Not Happen™ here, I'll add a warning and a comment similar to the sssd case.